File: 第10章/10-4 物件組合/Project1/Student.cpp
#include "Student.h"


Student::Student()
{
	name = "";
	score = 1;
}

Student::Student(string newName, Date newBirthday, int newScore)
{
	name = newName;
	birthDay = newBirthday;
	score = newScore;
}

void Student::setName(string newName)
{
	name = newName;
}

void Student::setDate(Date newBirthday)
{
	birthDay = newBirthday;
}

void Student::setScore(int newScore)
{
	score = newScore;
}

string Student::getName() const
{
	return name;
}

Date Student::getDate() const
{
	return birthDay;
}

int Student::getScore() const
{
	return score;
}

void Student::print()
{
	cout << name << " ";
	birthDay.print();
	cout << " " << score << endl;
}

File: python/StudentCard_Composition/Student.py
from Date import Date

class Student:

    def __init__(self, name, date, score):
        self.__name = name
        self.__date = date
        self.__score = score

    def setName(self, name):
        self.__name = name 

    def setDate(self, date):
        self.__date = date

    def setScore(self, score):
        self.__score = score

    def getName(self):
        return self.__name 

    def getDate(self):
        return self.__date 

    def getScore(self):
        return self.__score

    def toString(self):
        print(self.getName(), " ", self.getDate().toString(), " ", self.getScore())